openshift/deployment-template.yml

```diff
@@ -0,0 +1,121 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: go-qontract-reconcile
+objects:
+- apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    labels:
+      app: go-qontract-reconcile-${INTEGRATION_NAME}
+    annotations:
+      ignore-check.kube-linter.io/minimum-three-replicas: "go-qontract-reconcile integrations do not support replication"
+      ignore-check.kube-linter.io/no-readiness-probe: "go-qontract-reconcile integrations are not getting traffic"
+      ignore-check.kube-linter.io/no-liveness-probe: "go-qontract-reconcile integrations are monitored for being stuck"
+    name: go-qontract-reconcile-${INTEGRATION_NAME}
+  spec:
+    replicas: 1
+    strategy:
+      type: RollingUpdate
+      rollingUpdate:
+        maxUnavailable: 0
+        maxSurge: 1
+    selector:
+      matchLabels:
+        app: go-qontract-reconcile-${INTEGRATION_NAME}
+    template:
+      metadata:
+        labels:
+          app: go-qontract-reconcile-${INTEGRATION_NAME}
+          component: go-qontract-reconcile
+      spec:
+        serviceAccountName: go-qontract-reconcile
+        initContainers:
+        - name: internal-certificates
+          image: ${INTERNAL_CERTIFICATES_IMAGE}:${INTERNAL_CERTIFICATES_IMAGE_TAG}
+          imagePullPolicy: ${INTERNAL_CERTIFICATES_IMAGE_PULL_POLICY}
+          command: ["/bin/sh", "-c"]
+          args:
+          - |
+            cp -r /etc/pki/. /tmp/etc-pki/
+          volumeMounts:
+          - name: internal-certificates
+            mountPath: /tmp/etc-pki/
+        containers:
+        - image: ${IMAGE}:${IMAGE_TAG}
```

Comment:
how is slack logging performed? directly within go logger ? no need for a sidecar fluentd?

Reply:
Would opt for side car deployment, as in qontract-reconcile if it was required. However I'm reluctant to add all these features to the template. I'd rather go for integration go-qontract-reconcile with deployment manager.